### PR TITLE
Few fixes for Email field

### LINKED ISSUE #450: Field type 'Email' doesn't work with addFields();
When an 'Email' field type is added within addFields(), it is ignored. This might be a problem for other new fields as well but I haven't tested.

```
// Works
$model->addField('email', ['Email']);

// Doesn't work
$model->addFields([['email', ['Email']]]);
```

### LINKED ISSUE #471: Field type 'Email' stores NULL as empty string
When a NULL value is passed to a field of type 'Email', it seems to be stored as an empty string instead of NULL value.

### PR DESCRIPTION
**Pull request by DarkSide666**
fix #471 
fix #450

### CODE REVIEW

File: src/Field/Email.php [hunk 1]
```diff
@@ -49,6 +49,10 @@ class Email extends Field
      */
     public function normalize($value)
```
Comment (mvorisek) [inline src/Field/Email.php:50]:
```suggestion
    public function normalize($value): ?string
```

File: tests/FieldTypesTest.php [hunk 2]
```diff
@@ -31,21 +31,45 @@ public function testEmailBasic()
...
+        $m->save();
+        $this->assertNull($m->get('email'));
+
+        // normal value
         $m->set('email', 'foo@example.com');
         $m->save();
         $this->assertSame('foo@example.com', $m->get('email'));
 
-        // padding removed
+        // padding, spaceing etc removed
```
Comment (mvorisek) [inline tests/FieldTypesTest.php]:
```suggestion
        // padding, spacing etc removed
```

File: tests/FieldTypesTest.php [hunk 3]
```diff
@@ -31,21 +31,45 @@ public function testEmailBasic()
...
         $this->assertSame('foo@example.com', $m->get('email'));
 
+        // no domain - go to hell :)
         $this->expectException(ValidationException::class);
         $this->expectExceptionMessage('does not have domain');
         $m->set('email', 'qq');
     }
 
-    public function testEmailMultiple()
+    public function testMultipleEmailFields()
```
Comment (mvorisek) [inline tests/FieldTypesTest.php]: I do not think we need this test

File: src/Field/Email.php (hunk 1 above)
Comment (DarkSide666) [reply src/Field/Email.php:50]: Parent Field class do not have return value type set. Will that be OK to set it here?

File: tests/FieldTypesTest.php (hunk 3 above)
Comment (DarkSide666) [reply tests/FieldTypesTest.php]: I added this test here to show how it's working (see https://github.com/atk4/data/issues/450).
But probably you're right that this do not solve anything, just adds another test to prove that #450 can be closed :)

Comment (mvorisek) [reply tests/FieldTypesTest.php]: addFields is absolutely unrelated with email, I do not think we should test it, it also add LOC to maintain

**Pull request merged**